shared/js/telemetry.js

```diff
@@ -135,13 +134,45 @@
           }).then(function(result) {
             return result.statusMessage;
           }).then(function (imei) {
-            resolve(imei);
+            if (isDogfooder) {
+              resolve(imei);
+            } else {
+              return TelemetryRequest.createHash(imei);
+            }
+          }).then(function (hashed) {
+            resolve(hashed);
           }).catch(reject);
         }
       };
     });
   };
 
+    TelemetryRequest.createHash = function createHash(id) {
+      return new Promise(function (resolve, reject) {
+        var salt = new Uint8Array([44, 27, 51, 17, 55, 64, 183, 55, 37, 31, 246,
+          196, 130, 73, 14, 95, 122, 241, 143, 18, 40, 109,
+          238, 59, 168, 82, 28, 199, 177, 37, 135, 98]);
+        var hexDigits = '0123456789ABCDEF';
+        var ftuID = '';
+        var macString = navigator.mozWifiManager.macAddress.replace(/:/g, '');
+        var bytes = new TextEncoder('utf-8').encode(id + macString);
+        crypto.subtle.importKey('raw', bytes, {name:'PBKDF2'},
+                                false, ['deriveBits']).then(function (key) {
+          return crypto.subtle.deriveBits({
+            name: 'PBKDF2',
+            iterations: 1000,
+            salt: salt,
+            hash: 'SHA-1'
+          }, key, 160);
+        }).then(function (digest) {
+          var digestArray = new Uint8Array(digest).slice(0, 4);
```

Comment:
Unsure if it matters, but this is only ~4.2 billion unique IDs. maybe we should be more ambitious :)